src/librustc_trans/trans_item.rs

```diff
@@ -359,7 +338,7 @@ impl<'a, 'tcx> TransItem<'tcx> {
             TransItem::Fn(instance) => {
                 format!("Fn({:?}, {})",
                          instance.def,
-                         instance.substs as *const _ as usize)
+                         instance.substs.as_ptr() as usize)
```

Comment:
I take it back, the length is determined by `instance.def` so this chance is not necessary.